enterprise_access/apps/subsidy_access_policy/customer_api.py

```diff
@@ -0,0 +1,48 @@
+"""
+Python API for interacting with the lms client
+for use in the domain of SubsidyAccessPolicies.
+"""
+import logging
+
+from django.conf import settings
+from edx_django_utils.cache import TieredCache
+from requests.exceptions import HTTPError
+
+from enterprise_access.apps.api_client.lms_client import LmsApiClient
+from enterprise_access.cache_utils import versioned_cache_key
+
+logger = logging.getLogger(__name__)
+
+DEFAULT_CACHE_TIMEOUT = getattr(settings, 'LMS_CLIENT_CACHE_TIMEOUT', 60 * 5)
+
+
+def get_and_cache_enterprise_learner_record(enterprise_customer_uuid, learner_id, timeout=DEFAULT_CACHE_TIMEOUT):
+    """
+    Determines if the learner identified by the 'learner_id is linked
+    to a specified enterprise identified by the 'enterprise_customer_uuid
+
+    Returns: None or the enterprise customer user record
+    """
+
+    cache_key = versioned_cache_key('get_enterprise_contains_learner', enterprise_customer_uuid, learner_id)
+    cached_response = TieredCache.get_cached_response(cache_key)
+    if cached_response.is_found:
+        logger.info(f'Cache hit for customer {enterprise_customer_uuid} and learner id {learner_id}')
+        return cached_response.value
+
+    lms_client = LmsApiClient()
+    try:
+        is_learner_linked_to_enterprise = lms_client.get_enterprise_user(
```

Comment:
Nit: this variable name could be a little confusing, `get_enterprise_user` returns either a dict of customer-user data, or `None`. And you want this method we're in right now to return the same.